src/smtp.rs

```diff
@@ -69,13 +69,26 @@ impl Smtp {
         let domain = &lp.send_server;
         let port = lp.send_port as u16;
 
-        let tls = native_tls::TlsConnector::builder()
-            // see also: https://github.com/deltachat/deltachat-core-rust/issues/203
-            .danger_accept_invalid_hostnames(true)
-            .danger_accept_invalid_certs(true)
-            .min_protocol_version(Some(DEFAULT_TLS_PROTOCOLS[0]))
-            .build()
-            .unwrap();
+        let mut tls_builder = native_tls::TlsConnector::builder();
+        let tls = match lp.send_certificate_checks {
+            CertificateChecks::Automatic => {
+                // Same as AcceptInvalidCertificates for now.
+                // TODO: use provider database when it becomes available
+                tls_builder
+                    .danger_accept_invalid_hostnames(true)
+                    .danger_accept_invalid_certs(true)
+            }
+            CertificateChecks::Strict => &mut tls_builder,
+            CertificateChecks::AcceptInvalidHostnames => {
+                tls_builder.danger_accept_invalid_hostnames(true)
+            }
+            CertificateChecks::AcceptInvalidCertificates => tls_builder
+                .danger_accept_invalid_hostnames(true)
+                .danger_accept_invalid_certs(true),
+        }
+        .min_protocol_version(Some(DEFAULT_TLS_PROTOCOLS[0]))
```

Comment:
Why do we have `.min_protocol_version` for SMTP but not IMAP? It is inconsistent.

Reply:
@dignifiedquire If there is no particular reason you added this, I guess it is better to remove this line.

Reply:
Looks like it is equal to TLSv1.2 since https://github.com/lettre/lettre/commit/bab4519baa08af9d6d587ec7ce6e8c92bb37ed33

Maybe just set it to `Some(Protocol::Tlsv12)` manually both for IMAP and SMTP?

Reply:
yes, probably  it's better to be explicit and control it ourselves here.